code/modules/mob/living/basic/lavaland/gutlunchers/gutlunchers.dm

```diff
@@ -0,0 +1,172 @@
+		post_birth = CALLBACK(src, PROC_REF(after_birth)),\
+		breed_timer = 3 MINUTES,\
+	)
+
+/mob/living/basic/mining/gutlunch/proc/pre_attack(mob/living/puncher, atom/target)
+	SIGNAL_HANDLER
+
+	if(!istype(target, /obj/structure/ore_container/gutlunch_trough))
+		return
+
+	var/obj/ore_food = locate(/obj/item/stack/ore) in target
+
+	if(isnull(ore_food))
+		balloon_alert(src, "no food!")
+	else
+		melee_attack(ore_food)
+	return COMPONENT_HOSTILE_NO_ATTACK
+
+/mob/living/basic/mining/gutlunch/proc/after_birth(mob/living/basic/mining/gutlunch/grub/baby, mob/living/partner)
+	var/our_color = atom_colours[FIXED_COLOUR_PRIORITY] || COLOR_GRAY
+	var/partner_color = partner.atom_colours[FIXED_COLOUR_PRIORITY] || COLOR_GRAY
+	baby.add_atom_colour(BlendRGB(our_color, partner_color, 1), FIXED_COLOUR_PRIORITY)
+	var/atom/male_parent = (gender == MALE) ? src : partner
+	baby.inherited_stats = new(male_parent)
+
+/mob/living/basic/mining/gutlunch/proc/roll_stats(input_attack, input_speed, input_health)
+	melee_damage_lower = rand(input_attack, min(MAX_LOWER_ATTACK, input_attack + MAX_ATTACK_DIFFERENCE))
+	melee_damage_upper = melee_damage_lower + MAX_ATTACK_DIFFERENCE
+	speed = rand(MINIMUM_POSSIBLE_SPEED, input_speed)
+	maxHealth = rand(input_health, MAX_POSSIBLE_HEALTH)
+	health = maxHealth
+
+/mob/living/basic/mining/gutlunch/milk
+	name = "gubbuck"
+	gender = FEMALE
+	size_scale = 85
+	possible_colors = list("#E39FBB","#817178","#9d667d")
+	ai_controller = /datum/ai_controller/basic_controller/gutlunch/gutlunch_milk
+	///overlay we display when our udder is full!
+	var/mutable_appearance/full_udder
+
+/mob/living/basic/mining/gutlunch/milk/Initialize(mapload)
+	. = ..()
+	var/datum/callback/milking_callback = CALLBACK(src, TYPE_PROC_REF(/atom/movable, update_overlays))
+	AddComponent(\
+		/datum/component/udder,\
+		udder_type = /obj/item/udder/gutlunch,\
+		on_milk_callback = milking_callback,\
+		on_generate_callback = milking_callback,\
+	)
+	full_udder = mutable_appearance(icon, "gl_full")
+	full_udder.color = atom_colours[FIXED_COLOUR_PRIORITY] || COLOR_GRAY
+
+/mob/living/basic/mining/gutlunch/warrior
+	name = "gunther"
+	gender = MALE
+	size_scale = 100
+	melee_damage_lower = 8
+	melee_damage_upper = 13
+	speed = 5
+	maxHealth = 70
+	health = 70
+	ai_controller = /datum/ai_controller/basic_controller/gutlunch/gutlunch_warrior
+	possible_colors = list("#6d77ff","#8578e4","#97b6f6")
+	//pet commands when we tame the gutluncher
+	var/static/list/pet_commands = list(
+		/datum/pet_command/idle,
+		/datum/pet_command/free,
+		/datum/pet_command/point_targetting/attack,
+		/datum/pet_command/point_targetting/breed,
+		/datum/pet_command/follow,
+		/datum/pet_command/point_targetting/fetch,
+		/datum/pet_command/mine_walls,
+	)
+
+/mob/living/basic/mining/gutlunch/warrior/Initialize(mapload)
+	. = ..()
+	roll_stats(melee_damage_lower, speed, maxHealth)
+	AddComponent(/datum/component/obeys_commands, pet_commands)
+	AddElement(/datum/element/wall_tearer)
```

Comment:
```suggestion
	AddElement(/datum/element/wall_tearer, allow_reinforced = FALSE)
```
Just preemptively removing fun from ashwalkers who want to get onto the station.